### PR TITLE
Patch to go related modules to address CVE-2021-44716

### PR DESCRIPTION
**Pull request by nicogbg**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Added patch to address CVE-2021-44716 for go related modules.

###### Change Log  

- Patch CVE-2021-44716

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2021-44716

###### Test Methodology

- Pipeline build id: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=504232&view=results

### CODE REVIEW

**Review by tobiasb-ms — APPROVED** [review]

File: SPECS/application-gateway-kubernetes-ingress/application-gateway-kubernetes-ingress.spec [hunk 1]
```diff
@@ -24,9 +24,13 @@ Source0:        %{name}-%{version}.tar.gz
...
 #
 Source1:        %{name}-%{version}-vendor.tar.gz
+
+# patches for vendored code >= 1000
 # If upstream ever upgrades client_goland to 1.11.1, we can get rid of this patch.
-Patch0:         CVE-2022-21698.patch
-Patch1:         CVE-2023-44487.patch
+Patch1000:      CVE-2022-21698.patch
+Patch1001:      CVE-2023-44487.patch
+Patch1002:      CVE-2021-44716.patch
```
Comment (dmcilvaney) [inline SPECS/application-gateway-kubernetes-ingress/application-gateway-kubernetes-ingress.spec:32]: There might have been a sequencing issue, I removed the >= 1000 numbering on most golang packages.
```suggestion
Patch0:         CVE-2022-21698.patch
Patch1:         CVE-2023-44487.patch
Patch2:         CVE-2021-44716.patch
```

File: SPECS/application-gateway-kubernetes-ingress/application-gateway-kubernetes-ingress.spec [hunk 2]
```diff
@@ -24,9 +24,13 @@ Source0:        %{name}-%{version}.tar.gz
 #           -cf %%{name}-%%{version}-vendor.tar.gz vendor
 #
 Source1:        %{name}-%{version}-vendor.tar.gz
+
+# patches for vendored code >= 1000
```
Comment (dmcilvaney) [inline SPECS/application-gateway-kubernetes-ingress/application-gateway-kubernetes-ingress.spec:28]:
```suggestion
```

File: SPECS/cf-cli/cf-cli.spec [hunk 3]
```diff
@@ -27,7 +27,10 @@ Source0:        https://github.com/cloudfoundry/cli/archive/refs/tags/v%{version
 #         See: https://reproducible-builds.org/docs/archives/
 #       - For the value of "--mtime" use the date "2021-04-26 00:00Z" to simplify future updates.
 Source1:        cli-%{version}-vendor.tar.gz
-Patch0:         CVE-2023-44487.patch
+
+# patches for vendored code >= 1000
+Patch1000:      CVE-2023-44487.patch
+Patch1001:      CVE-2021-44716.patch
```
Comment (dmcilvaney) [inline SPECS/cf-cli/cf-cli.spec:33]:
```suggestion
Patch0:         CVE-2023-44487.patch
Patch1:         CVE-2021-44716.patch
```

File: SPECS/csi-driver-lvm/csi-driver-lvm.spec [hunk 4]
```diff
@@ -19,6 +19,10 @@ Source0:        https://github.com/metal-stack/%{name}/archive/refs/tags/v%{vers
 #           --pax-option=exthdr.name=%d/PaxHeaders/%f,delete=atime,delete=ctime \
 #           -cf %%{name}-%%{version}-govendor.tar.gz vendor
 Source1:        %{name}-%{version}-govendor.tar.gz
+
+# patches for vendored code >= 1000
+Patch1000:      CVE-2021-44716.patch
```
Comment (dmcilvaney) [inline SPECS/csi-driver-lvm/csi-driver-lvm.spec:24]:
```suggestion
Patch0:         CVE-2021-44716.patch
```

File: SPECS/git-lfs/git-lfs.spec [hunk 5]
```diff
@@ -28,7 +28,10 @@ Source0:       https://github.com/git-lfs/git-lfs/archive/v%{version}.tar.gz#/%{
 #         See: https://reproducible-builds.org/docs/archives/
 #       - For the value of "--mtime" use the date "2021-04-26 00:00Z" to simplify future updates.
 Source1:       %{name}-%{version}-vendor.tar.gz
-Patch0:        CVE-2023-44487.patch
+
+# patches for vendored code >= 1000
+Patch1000:     CVE-2023-44487.patch
+Patch1001:     CVE-2021-44716.patch
```
Comment (dmcilvaney) [inline SPECS/git-lfs/git-lfs.spec:34]:
```suggestion
Patch0:        CVE-2023-44487.patch
Patch1:        CVE-2021-44716.patch
```

File: SPECS/jx/jx.spec [hunk 6]
```diff
@@ -27,7 +27,10 @@ Source0:        https://github.com/jenkins-x/jx/archive/v%{version}.tar.gz#/%{na
 #         See: https://reproducible-builds.org/docs/archives/
 #       - For the value of "--mtime" use the date "2021-04-26 00:00Z" to simplify future updates.
 Source1:        %{name}-%{version}-vendor.tar.gz
-Patch0:         CVE-2023-44487.patch
+
+# patches for vendored code >= 1000
+Patch1000:      CVE-2023-44487.patch
+Patch1001:      CVE-2021-44716.patch
```
Comment (dmcilvaney) [inline SPECS/jx/jx.spec:33]:
```suggestion
Patch0:         CVE-2023-44487.patch
Patch1:         CVE-2021-44716.patch
```

File: SPECS/keda/keda.spec [hunk 7]
```diff
@@ -29,7 +29,10 @@ Source1:        %{name}-%{version}-vendor-v2.tar.gz
 # Patches the version of client_golang used in the vendored source. Should be applied before creating the vendored tarball.
 # Can be removed if we upgrade keda to 2.6.0 or later.
 Patch0:         CVE-2022-21698.patch
-Patch1:         CVE-2023-44487.patch
+
+# patches for vendored code >= 1000
+Patch1000:         CVE-2023-44487.patch
+Patch1001:         CVE-2021-44716.patch
```
Comment (dmcilvaney) [inline SPECS/keda/keda.spec:35]:
```suggestion
Patch0:            CVE-2023-44487.patch
Patch1:            CVE-2021-44716.patch
```

**Review by dmcilvaney — CHANGES_REQUESTED** [review]

**Review by dmcilvaney — APPROVED** [review]

**Pull request merged**